commonmark/src/main/java/org/commonmark/html/HtmlRenderer.java

```diff
@@ -148,8 +150,18 @@ public Builder attributeProvider(AttributeProvider attributeProvider) {
             return this;
         }
 
-        public Builder customHtmlRenderer(CustomHtmlRenderer customHtmlRenderer) {
-            this.customHtmlRenderers.add(customHtmlRenderer);
+        /**
+         * Add a factory for instantiating a node renderer (done when rendering). This allows to override the rendering
+         * of node types or define rendering for custom node types.
+         * <p>
+         * If multiple node renderers for the same node type are created, the one from the factory that was added first
+         * "wins". (This is how the rendering for core node types can be overridden; the default rendering comes last.)
+         *
+         * @param nodeRendererFactory the factory for creating a node renderer
+         * @return {@code this}
+         */
+        public Builder nodeRendererFactory(NodeRendererFactory nodeRendererFactory) {
```

Comment:
This is the new extension point. See also [NodeRenderer](https://github.com/atlassian/commonmark-java/pull/49/files#diff-f6dfcbbe0c786ca98aae0f5a8c8e6498) and associated interfaces.